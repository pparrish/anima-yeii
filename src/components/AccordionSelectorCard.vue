<template>
  <div class="card">
    <div class="card-header">
      <div class="card-header-title">
        <p>{{ title }} 
          <span
            v-for="(tag, index) in tags"
            :key="index"
            class="tag">{{ tag }}</span></p>
      </div>
      <a class="card-header-icon">
        <span class="icon">
          <i class="fas fa-angle-down"/>
        </span>
      </a>
    </div>
    <div class="card-content has-justified-text">
      <p>{{ content }}</p>
    </div>
    <div class="card-footer">
      <a 
        class="card-footer-item button is-link is-radiusless"
        @click="$emmit('selected') ">{{ buttonText }}</a>
    </div>
  </div>
</template>

<script>
export default {
  name: "AccordeonSelectorCard",
  props: {
    title: {
      type: String,
      default: () => "No title",
      required: false
    },
    tags: {
      type: Array,
      default: () => [],
      required: false
    },
    content: {
      type: String,
      default: () => "No content",
      required: false
    },
    buttonText: {
      type: String,
      default: () => "No text",
      required: false
    }
  }
};
</script>
