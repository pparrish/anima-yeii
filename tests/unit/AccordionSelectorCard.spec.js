import { shallowMount } from "@vue/test-utils";
import AccordionSelectorCard from "@/components/AccordionSelectorCard";

describe("AccordionSelectorCard.vue", () => {
  let cmp;

  beforeEach(() => {
    cmp = shallowMount(AccordionSelectorCard);
  });

  it("has the expected html structure", () => {
    expect(cmp.vm.$el).toMatchSnapshot();
  });

  describe("All default values is set default correctly", () => {
    it("default title must be 'No title'", () => {
      expect(cmp.vm.title).toEqual("No title");
    });
    it("default tagg must be []", () => {
      expect(cmp.vm.tags).toEqual([]);
    });
    it("default content must be 'No content'", () => {
      expect(cmp.vm.content).toEqual("No content");
    });
    it("default title must be 'No text'", () => {
      expect(cmp.vm.buttonText).toEqual("No text");
    });
  });

  describe("Props must be  set correctly", () => {
    it("Title must be 'Selector Card'", () => {
      cmp = shallowMount(AccordionSelectorCard, {
        propsData: {
          title: "Selector Card"
        }
      });

      expect(cmp.vm.title).toBe("Selector Card");
    });

    it("tags must be ['tag', 'a tag']", () => {
      cmp = shallowMount(AccordionSelectorCard, {
        propsData: {
          tags: ["tag", "a tag"]
        }
      });

      expect(cmp.vm.tags).toEqual(["tag", "a tag"]);
    });

    it("content must be 'A content'", () => {
      cmp = shallowMount(AccordionSelectorCard, {
        propsData: {
          content: "A content"
        }
      });

      expect(cmp.vm.content).toEqual("A content");
    });

    it("Button must be say 'Select me'", () => {
      cmp = shallowMount(AccordionSelectorCard, {
        propsData: {
          buttonText: "Select me"
        }
      });

      expect(cmp.vm.buttonText).toEqual("Select me");
    });

    it("Haz the espected structure", () => {
      expect(cmp.vm.$el).toMatchSnapshot();
    });
  });
});
