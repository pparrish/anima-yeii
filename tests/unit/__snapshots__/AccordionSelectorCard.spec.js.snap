// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AccordionSelectorCard.vue Props must be  set correctly Haz the espected structure 1`] = `
<div
  class="card"
>
  <div
    class="card-header"
  >
    <div
      class="card-header-title"
    >
      <p>
        No title 
        
      </p>
    </div>
     
    <a
      class="card-header-icon"
    >
      <span
        class="icon"
      >
        <i
          class="fas fa-angle-down"
        />
      </span>
    </a>
  </div>
   
  <div
    class="card-content has-justified-text"
  >
    <p>
      No content
    </p>
  </div>
   
  <div
    class="card-footer"
  >
    <button
      class="card-footer-item button is-link is-radiusless"
    >
      No text
    </button>
  </div>
</div>
`;

exports[`AccordionSelectorCard.vue has the expected html structure 1`] = `
<div
  class="card"
>
  <div
    class="card-header"
  >
    <div
      class="card-header-title"
    >
      <p>
        No title 
        
      </p>
    </div>
     
    <a
      class="card-header-icon"
    >
      <span
        class="icon"
      >
        <i
          class="fas fa-angle-down"
        />
      </span>
    </a>
  </div>
   
  <div
    class="card-content has-justified-text"
  >
    <p>
      No content
    </p>
  </div>
   
  <div
    class="card-footer"
  >
    <button
      class="card-footer-item button is-link is-radiusless"
    >
      No text
    </button>
  </div>
</div>
`;
